File: frameworks/test.container/src/test/java/com/everest/frameworks/test/container/DummyTestContainer.java
package com.everest.frameworks.test.container;

import org.jetbrains.annotations.NotNull;
import org.testcontainers.lifecycle.Startable;

public class DummyTestContainer extends TestContainer<Startable> {

    static int numberOfStarts;
    static int numberOfEnvironmentSetups;

    public DummyTestContainer() {
        super();
    }

    @Override
    protected @NotNull Startable createContainer() {
        return new Startable() {
            @Override
            public void start() {
                numberOfStarts += 1;
            }

            @Override
            public void stop() {
            }
        };
    }

    @Override
    protected void setupEnvironment() {
        numberOfEnvironmentSetups += 1;
    }

}

File: frameworks/test.container/src/test/kotlin/com/qomolangma/frameworks/test/container/DummyTestContainer.kt
package com.qomolangma.frameworks.test.container

import org.testcontainers.lifecycle.Startable

class DummyTestContainer : TestContainer<Startable>() {
    override fun createContainer(): Startable {
        return object : Startable {
            override fun start() {
                numberOfStarts += 1
            }

            override fun stop() {}
        }
    }

    override fun setupEnvironment() {
        numberOfEnvironmentSetups += 1
    }

    companion object {
        @JvmField
        var numberOfStarts = 0

        @JvmField
        var numberOfEnvironmentSetups = 0
    }
}
